### PR TITLE
20417 cookie attribute

### PR DESCRIPTION
**Pull request by tbitonti**
Parsing the web.xml for new cookie-config's attribute element

https://github.com/OpenLiberty/open-liberty/issues/20417

### CODE REVIEW

**Comment by tbitonti** [comment]
#build

**Comment by tbitonti** [comment]
Testing (included in the PR):

com.ibm.ws.javaee.ddmodel.web.WebAppTest
- testEE10CookieAttribute_Missing()  - No attribute element
- testEE10CookieAttributeNoDescriptions() - 1 attribute with no description elements
- testEE10CookieAttributeOneDescription() - 1 attribute with 1 description
- testEE10CookieAttributeTwoDescriptions() - 1 attribute with 2 descriptions
- testEE10CookieAttributeMultiple()  - 2 attributes with 2 descriptions

File: dev/com.ibm.ws.javaee.ddmodel/test/com/ibm/ws/javaee/ddmodel/web/WebAppTest.java [hunk 1]
```diff
@@ -177,167 +180,363 @@ public void testEE6Web30DenyUncoveredHttpMethods() throws Exception {
...
+
+        List<Description> descriptions = attribute.getDescriptions();
+        Description description = descriptions.get(0);
+        Assert.assertEquals(description.getLang(), "en");
+        Assert.assertEquals(description.getValue(), "Hair color");
+
+        description = descriptions.get(1);
+        Assert.assertEquals(description.getLang(), "fr");
+        Assert.assertEquals(description.getValue(), "Couleur de cheveux");
+
```
Comment (pmd1nh) [inline dev/com.ibm.ws.javaee.ddmodel/test/com/ibm/ws/javaee/ddmodel/web/WebAppTest.java:476]: Should attribute-name also be tested?

File: dev/com.ibm.ws.javaee.ddmodel/test/com/ibm/ws/javaee/ddmodel/web/WebAppTest.java [hunk 2]
```diff
@@ -177,167 +180,363 @@ public void testEE6Web30DenyUncoveredHttpMethods() throws Exception {
...
+        //     first description
+        description = descriptions.get(0);
+        Assert.assertEquals(description.getLang(), "en");
+        Assert.assertEquals(description.getValue(), "Facial complecxion");
+
+        //     second description
+        description = descriptions.get(1);
+        Assert.assertEquals(description.getLang(), "fr");
+        Assert.assertEquals(description.getValue(), "Complexión facial");
```
Comment (pmd1nh) [inline dev/com.ibm.ws.javaee.ddmodel/test/com/ibm/ws/javaee/ddmodel/web/WebAppTest.java:546]: Same here...should attribute-name and value pair be tested?

**Comment by tbitonti** [comment]
#build

**Comment by tbitonti** [comment]
#libby

File: dev/com.ibm.ws.javaee.ddmodel/test/com/ibm/ws/javaee/ddmodel/web/WebAppTest.java (hunk 2 above)
Comment (tbitonti) [reply dev/com.ibm.ws.javaee.ddmodel/test/com/ibm/ws/javaee/ddmodel/web/WebAppTest.java:546]: Added tests on attribute-name.

File: dev/com.ibm.ws.javaee.ddmodel/test/com/ibm/ws/javaee/ddmodel/web/WebAppTest.java (hunk 1 above)
Comment (tbitonti) [reply dev/com.ibm.ws.javaee.ddmodel/test/com/ibm/ws/javaee/ddmodel/web/WebAppTest.java:476]: Added tests on attribute name.

**Comment by tbitonti** [comment]
First try at resolving the version checker problems ...

**Comment by tbitonti** [comment]
No version checker errors.  A number of test failures which seem unrelated:

- com.ibm.ws.jaxrs.2.1_fat FAT tests
  - com.ibm.ws.jaxrs21.fat.JAXRS21SecurityAnnotationsTest
    - com.ibm.ws.ui.tool.explore_fat.gui.nd.rw.ws.allresources FAT tests
  - com.ibm.ws.ui.tool.explore.fat.gui.nd.rw.allresources.tests.AllClustersTest
    - testClusterSearchViewActionBarTest_withWebSocket
- io.openliberty.jcache.internal_fat.infinispan FAT tests
  - io.openliberty.jcache.internal.fat.JCacheDeleteAuthCacheTest
    - deleteAuthCache_mbean_EE9_FEATURES

**Review by pmd1nh — APPROVED** [review]

**Pull request merged**